app/src/main/java/ch/epfl/culturequest/ui/profile/ProfileViewModel.java

```diff
@@ -30,33 +31,33 @@ public ProfileViewModel(String uid) {
 
         EspressoIdlingResource.increment();
         Profile profile = Profile.getActiveProfile();
+        Profile selectedProfile = ProfileUtils.getSelectedProfile();
 
         if (profile != null) {
-            //set the values of the live data
-            username.setValue(profile.getUsername());
-            profilePictureUri.setValue(profile.getProfilePicture());
-            pictures.setValue(profile.getImagesList());
-
-
-            // add an observer to the profile so that the view is updated when the profile is updated
-            profile.addObserver((profileObject, arg) -> {
-                Profile p = (Profile) profileObject;
-                username.postValue(p.getUsername());
-                profilePictureUri.postValue(p.getProfilePicture());
-                pictures.postValue(p.getImagesList());
-            });
-
+            if (selectedProfile != null
+                    && selectedProfile.getUid().equals(uid)) {
+                Database.getProfile(uid).whenComplete((p, e) -> {
+                    username.setValue(p.getUsername());
+                    profilePictureUri.setValue(p.getProfilePicture());
+                    pictures.setValue(p.getImagesList());
+                });
```

Comment:
why do you need to fetch the profile from the database if you already have selectedProfile?